src/cmap/connection_pool.ts

```diff
@@ -383,6 +450,50 @@ export class ConnectionPool extends TypedEventEmitter<ConnectionPoolEvents> {
       });
     });
   }
+
+  /**
+   * Runs a lambda with an implicitly checked out connection, checking that connection back in when the lambda
+   * has completed by calling back.
```

Comment:
This is looking like the `withConnection` description above, and maybe I’m missing something, but where’s the `checkIn()` call here?

Reply:
I updated the comments now to reflect what this method really does.